src/index.js

```diff
@@ -313,6 +296,43 @@ follow these steps, YOUR APP MAY BREAK.`);
     Firestore.log('Firestore', null, 'Initialized Firestore');
   }
 
+  /**
+   * Specifies custom settings to be used to configure the `Firestore`
+   * instance. Can only be invoked once and before any other Firestore method.
+   *
+   * If settings are provided via both `settings()` and the Firestore
+   * constructor, both settings objects are merged and any settings provided via
+   * `settings()` take precedence.
+   *
+   * @param {object} settings The settings to use for all Firestore operations.
+   */
+  settings(settings) {
+    validate.isObject('settings', settings);
+    validate.isOptionalString('settings.projectId', settings.projectId);
+    validate.isOptionalBoolean(
+        'settings.timestampsInSnapshots', settings.timestampsInSnapshots);
+
+    if (this._clientInitialized) {
```

Comment:
May be start with these 2 checks, before validating the `settings` argument.

Reply:
Most other functions call these validation helpers right away, so I am inclined to leave it as is. Moving these helper calls further down also looks a little "goofy" to me.